tools/pyosmium-up-to-date

```diff
@@ -174,14 +193,24 @@ def get_arg_parser(from_main=False):
     parser.add_argument('--force-update-of-old-planet', action='store_true',
                         dest='force_update',
                         help="Apply update even if the input data is really old.")
+    parser.add_argument('--cookie', dest='cookie',
+                        help="Netscape-style cookie jar file to read cookies from")
 
     return parser
 
+def open_with_cookie(url):
+    return opener.open(url)
+
 if __name__ == '__main__':
     logging.basicConfig(stream=sys.stderr,
                         format='%(asctime)s %(levelname)s: %(message)s')
 
     options = get_arg_parser(from_main=True).parse_args()
+    # According to the documentation, the cookie jar loads the file only if FileCookieJar.load is called.
+    cookie_jar = cookiejarlib.MozillaCookieJar(options.cookie)
+    if options.cookie is not None:
+        cookie_jar.load(options.cookie)
+    opener = urlrequest.build_opener(urlrequest.HTTPCookieProcessor(cookie_jar))
```

Comment:
This should go completely into update_from_custom_server(). Same comment as above applies about only doing it when cookie is set.

Reply:
I did not put it into `update_from_custom_server()` because that function has multiple return statements. Calling a function which calls `cookie_jar.save(options.cookie)` from multiple locations in `update_from_custom_server()` looked like too much duplication for me.

Reply:
It has only one successful return as far as I can see. Before the final return is the only place where the cookie is actually saved.